python/cudf/cudf/dataframe/string.py

```diff
@@ -755,6 +755,8 @@ def normalize_binop_value(self, other):
             col = utils.scalar_broadcast_to(
                 other, shape=len(self), dtype="object"
             )
+            if isinstance(col, str):
+                col = StringColumn(nvstrings.to_device([col]))
```

Comment:
This should be handled in `utils.scalar_broadcast_to`, no?

Reply:
Because what's happening here is that you copy the string scalar to the device via nvstrings, it runs a `__getitem__` with an array-like of length 1 which triggers a copy back to host, and then you copy it back to device.